### PR TITLE
Nhi/region peeking

### PR DESCRIPTION
**Pull request by nhiquach**
### What does this PR do?
Allows a user to toggle the regions they are currently viewing. 

### Status
READY

### Description of Testing Done
Could use more thorough testing? Not sure what the full set of edge cases are here

### Todos
- [ ] Testing

### Screenshots
region-peeking

CC: @michaeldo2

### CODE REVIEW

File: helpers/LocalStorage.js [hunk 1]
```diff
@@ -44,6 +44,27 @@ class LocalStorage {
...
+        return Promise.resolve(JSON.parse(region_json));
+      } else {
+        return Promise.reject(new Error("No stored region"));
+      }
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  static clearRegion() {
```
Comment (michaeldo2) [inline helpers/LocalStorage.js]: When are we calling this?

File: helpers/LocalStorage.js [hunk 2]
```diff
@@ -44,6 +44,27 @@ class LocalStorage {
...
     AsyncStorage.removeItem(StorageKeys.userHasClaimed);
   }
+
+  static storeRegion(region_json) {
+    AsyncStorage.setItem(StorageKeys.region, JSON.stringify(region_json));
+  }
+
+  static async getRegion() {
+    try {
+      var region_json = await AsyncStorage.getItem(StorageKeys.region);
```
Comment (michaeldo2) [inline helpers/LocalStorage.js]: Asyncstorage will persist across app opening and closing right? Should we ever be bringing them back to their default region?

File: components/listings/ListingsTitle.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  _toggleModal = () => {
+    this.props.onPressItem();
+  }
+
+  render() {
+    return (
+      <View style={[styles.container]}>
+        <Button
+          onPress={this._toggleModal}>
+        <Text style={[styles.text, {color: Colors.white}]}>
```
Comment (michaeldo2) [inline components/listings/ListingsTitle.js]: small thing i think this should be indented 1 more as well as Icon

File: components/listings/ListingsTitle.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  static defaultProps = {
+    iconName: downCaret
+  }
+  
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      iconName: this.props.iconName,
```
Comment (michaeldo2) [inline components/listings/ListingsTitle.js]: Is iconName ever changing? Otherwise we should just be able to use it as a prop in the render() function instead of setting it into state

File: helpers/LocalStorage.js (hunk 1 above)
Comment (nhiquach) [reply helpers/LocalStorage.js]: lmao forgot to delete all this stuff
